mllib/src/main/scala/org/apache/spark/ml/clustering/KMeans.scala

```diff
@@ -303,6 +312,20 @@ class KMeans @Since("1.5.0") (
   @Since("1.5.0")
   def setSeed(value: Long): this.type = set(seed, value)
 
+  /** @group setParam */
```

Comment:
I don't feel really strongly about it. But I think it is not a critical failure, so does not really require a hard stop of an exception. Since we make it clear in the doc that initial model `k` takes precedence, if a user sets a different `k` anyway we can still proceed, just let them know that we are ignoring their setting.